src/MCPClient/lib/clientScripts/restructureDIPForContentDMUpload.py

```diff
@@ -640,569 +144,115 @@ def addAipUuidToDcMetadata(dipUuid, dcMetadata):
-                        delimChildValuesRow.insert(0, v['label'])
-                        for i in range(1, len(delimHeaderRow) - 1):
-                            delimChildValuesRow.append('')
-                    # Rows for single compound itms must contain filenamein the last position.
-                    delimChildValuesRow.append(childFilename)
-                    writer.writerow(delimChildValuesRow)
-               
-    delimitedFile.close()
-
+        csv_path = os.path.join(output_dir, 'simple.txt')
+
+    print 'package_type', package_type
+    print 'csv_path', csv_path
+
+    divs_with_dmdsecs = structmap.findall('.//mets:div[@DMDID]', namespaces=ns.NSMAP)
+    with open(csv_path, "wb") as csv_file:
+        writer = csv.writer(csv_file, delimiter='\t')
+
+        # Iterate through every div and create a row for each
+        csv_header_ref = None
+        for div in divs_with_dmdsecs:
+            # Find associated dmdSecs
+            dmdids = div.get('DMDID').split()
+            # Take nonDC dmdSec, fallback to DC dmdSec
+            dmdsecpair = splitDmdSecs([dmdsecs[dmdid] for dmdid in dmdids])
+            dmdsecpair['dc'] = addAipUuidToDcMetadata(dipuuid, dmdsecpair['dc'])
+            metadata = dmdsecpair['nonDc'] or dmdsecpair['dc']
+            # Create csv_header and csv_values from the dmdSec metadata
+            csv_header = []
+            csv_values = []
+            for header, value in metadata.iteritems():
+                csv_header.append(header)
+                value = '; '.join(value).replace('\r', '').replace('\n', '')
+                csv_values.append(archivematicaFunctions.unicodeToStr(value))
+
+            # Add AIP UUID
+            csv_header.append('AIP UUID')
+            csv_values.append(dipuuid)
+
+            # Add file UUID
+            csv_header.append('file UUID')
+            if 'dirs' in package_type:
+                # Directories have no file UUID
+                csv_values.append('')
+            else:
+                file_uuid = ''
+                fptr = div.find('mets:fptr', namespaces=ns.NSMAP)
+                # Only files have fptrs as direct children
+                if fptr is not None:
+                    # File UUID is last 36 characters of FILEID
+                    file_uuid = fptr.get('FILEID')[-36:]
+                csv_values.append(file_uuid)
+
+            # Add file or directory name
+            name = div.attrib['LABEL']  # Fallback if LABEL doesn't exist?
+            if 'dirs' in package_type:
+                csv_header.insert(0, 'Directory name')
+                csv_values.insert(0, name)
+            else:
+                csv_header.append('Filename')
+                csv_values.append(name)
+
+            # Compare csv_header, if diff ERROR (first time set, write to file)
+            if csv_header_ref and csv_header_ref != csv_header:
+                print >> sys.stderr, 'ERROR headers differ,', csv_path, 'almost certainly invalid'
```

Comment:
Discussed with Sarah, and looks like it should fail.  This case shouldn't happen, anyway.